src/cmd/board.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 The ChromiumOS Authors
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+use anyhow::bail;
+use anyhow::Result;
+use argh::FromArgs;
+use glob::Pattern;
+use lium::cache::KvCache;
+use lium::repo::get_repo_dir;
+use lium::util::shell_helpers::run_bash_command;
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// list supported boards
+#[argh(subcommand, name = "board")]
+pub struct Args {
+    #[argh(subcommand)]
+    nested: SubCommand,
+}
+
+pub static BOARD_CACHE: KvCache<Vec<String>> = KvCache::new("board_cache");
+static DEFAULT_BOARD_SET: &str = "board";
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand)]
+enum SubCommand {
+    List(ArgsList),
+}
+pub fn run(args: &Args) -> Result<()> {
+    match &args.nested {
+        SubCommand::List(args) => run_board_list(args),
+    }
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// List up the supporte boards
+#[argh(subcommand, name = "list")]
+pub struct ArgsList {
+    /// target cros repo directory
+    #[argh(option)]
+    repo: Option<String>,
+
+    /// glob pattern of the boards
+    #[argh(positional)]
+    boards: Option<String>,
```

Comment:
Maybe this should be names as "filter" or "query" or something (to avoid inconsistent use of "--board(s)" option compared to other commands)

Reply:
indeed. This is actually a filter so should be `filter`.